### PR TITLE
ENG-5120 mt add tenant config validation

### PR DESCRIPTION
**Pull request by MEM2677**
Added misc checks such as:
 - uniqueness of FQDN
 - uniqueness of the tenant code
 - cmDb* fields set
 - KC valid URL

### CODE REVIEW

File: src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java [hunk 1]
```diff
@@ -124,4 +128,58 @@ public static String validateEntityCodeOrThrow(String entityCode) throws Entando
...
+            final String port = String.valueOf(url.getPort());
+
+            return ((StringUtils.isNotBlank(url.toURI().toString())
+                    && StringUtils.isNotBlank(url.getProtocol()))
+                    && StringUtils.isNotBlank(url.getHost())
+                    && (!validatePort
+                    || (StringUtils.isNotBlank(port) && StringUtils.isNumeric(port)))
+                    && (!validatePath || StringUtils.isNotBlank(url.getPath())));
+        } catch (Throwable t) {
+            // no op
```
Comment (entgigi) [inline src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java]: pls put a log here, in general you should log more

File: src/main/java/org/entando/kubernetes/validator/TenantValidator.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            // check uniqueness of crucial settings
+            validateTenantIdUniqueness(tenants);
+            validateFqdnsUniqueness(tenants);
+        }
+    }
+
+    private void validateTenantIdUniqueness(List<TenantConfigDTO> tenants) {
+        final Map<String, Boolean> ids = new HashMap<>();
+
+        if (tenants != null && !tenants.isEmpty()) {
```
Comment (entgigi) [inline src/main/java/org/entando/kubernetes/validator/TenantValidator.java]: check not useful, already done before

File: src/main/java/org/entando/kubernetes/validator/TenantValidator.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            Arrays.asList(fqdns).forEach(fqdn -> {
+                if (!validateFQDN(fqdn)) {
+                    getErrorListForTenant(tenantCode).add("fqdns: invalid value detected '" + fqdn + "'");
+                }
+            });
+        }
+    }
+
+    private void validateFqdnsUniqueness(List<TenantConfigDTO> tenants) {
+        if (tenants != null && !tenants.isEmpty()) {
```
Comment (entgigi) [inline src/main/java/org/entando/kubernetes/validator/TenantValidator.java]: check not useful, already done before

File: src/main/java/org/entando/kubernetes/validator/TenantValidator.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        }
+        return validationErrors.get(id);
+    }
+
+    public Optional<Map<String, List<String>>> getValidationErrorMap() {
+        if (validationErrors != null
+                && !validationErrors.isEmpty()) {
+            validationErrors.keySet().forEach(k -> {
+                log.error("Tenant '{}'", k);
+                validationErrors.get(k).forEach(e -> log.error("\t{}", e));
```
Comment (entgigi) [inline src/main/java/org/entando/kubernetes/validator/TenantValidator.java]: you will not be sure the log will be in consecutive lines, you should put all in a unique log

**Review by entgigi — APPROVED** [review]
I've only added a few minor comments so in general, since you're in a rush, you can go ahead and merge it quickly, LGTM

File: src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java (hunk 1 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java:159]: Ok!

File: src/main/java/org/entando/kubernetes/validator/TenantValidator.java (hunk 2 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/validator/TenantValidator.java]: Ditto

File: src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java (hunk 1 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/validator/ValidationFunctions.java]: Done

File: src/main/java/org/entando/kubernetes/validator/TenantValidator.java (hunk 3 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/validator/TenantValidator.java]: Ok

File: src/main/java/org/entando/kubernetes/validator/TenantValidator.java (hunk 4 above)
Comment (MEM2677) [reply src/main/java/org/entando/kubernetes/validator/TenantValidator.java]: Nice catch, fixed!

**Pull request merged**